### ISSUE TITLE
KubeObjectListLayout filterItems regression

### ISSUE THREAD

**Issue by zhilinn** [lensapp/lens#2291]
**Describe the bug**
`<KubeObjectListLayout>` ignores `filterItems` property starting from 4.1.0-beta.2.

The regression seems to be introduced by #2095.
 
**To Reproduce**
```tsx
// should render an empty list, but instead it renders all the items it has in the store
<Component.KubeObjectListLayout
  store={someStore}
  filterItems={[items => []]}
  // ...
/>
```
This should be enough to reproduce unless I'm missing something. Will provide more info if needed.

**Comment by jim-docker** [comment]
We have a fix (https://github.com/lensapp/lens/pull/2241) but FYI the filterItems property will likely be deprecated some time soon. You could look at how events use the KubeObjectListLayout to do similar filtering. (events used to use filterItems): https://github.com/lensapp/lens/blob/e87bb850fb6525967fddce6218a0f77396f47970/src/renderer/components/%2Bevents/events.tsx#L128

**Comment by aleksfront** [comment]
Fixed in #2241
